odl/space/cu_ntuples.py

```diff
@@ -943,8 +1318,14 @@ def norm(self, x):
         norm : float
             The norm of the vector
         """
-        from math import sqrt
-        return sqrt(self.const) * float(_norm_default(x))
+        if self.exponent == 2.0:
+            from math import sqrt
+            return sqrt(self.const) * float(_norm_default(x))
+        elif self.exponent == float('inf'):  # Weighting irrelevant
```

Comment:
Hmm ok, but you could still define a norm as `max(x * w)`, which is probably what people would expect, right?